src/beamlime/core/config_subscriber.py

```diff
@@ -32,22 +33,25 @@ def start(self):
         self._thread = threading.Thread(target=self._run_loop)
         self._thread.start()
 
+    def _process_message(self, message: KafkaMessage | None) -> None:
+        if message is None:
+            return
+        if message.error():
+            self._logger.error('Consumer error: %s', message.error())
+            return
+        key = message.key().decode('utf-8')
+        value = json.loads(message.value().decode('utf-8'))
+        self._logger.info(
+            'Updating config: %s = %s at %s', key, value, message.timestamp()
+        )
+        self._config[key] = value
+
     def _run_loop(self):
         try:
             while self._running:
-                msg = self._consumer.poll(0.1)
-                if msg is None:
-                    continue
-                if msg.error():
-                    self._logger.error('Consumer error: %s', msg.error())
-                    continue
-
-                key = msg.key().decode('utf-8')
-                value = json.loads(msg.value().decode('utf-8'))
-                self._logger.info(
-                    'Updating config: %s = %s at %s', key, value, msg.timestamp()
-                )
-                self._config[key] = value
+                messages = self._consumer.consume(num_messages=100, timeout=0.1)
```

Comment:
Why does the config consumer need to consume so many messages at once?

Reply:
To enable better batching and minimize the number of back and forth with the broker? True, this usually does not happen as few values are updated, but it does when loading initial config or catching up after restart.

Do you think it is harmful?